src/examples/calculator.js

```diff
@@ -0,0 +1,255 @@
+import dedent from "dedent";
+
+export const CALCULATOR = {
+  title: "Calculator",
+  description: `This example is a small calculator app written in React, which
+  uses exact Decimal calculations instead of floating point. It shows how to
+  manipulate the DOM from the playground.`,
+  text: dedent`
+  // Make sure to click "Toggle DOM" above
+  // to see the app!
+
+  function isInteger(dec) {
+    const truncOptions = {
+      maximumFractionDigits: 0,
+      roundingMode: "down",
+    };
+    return Decimal.round(dec, truncOptions) === dec;
+  }
+
+  class Key extends React.Component {
+    render() {
+      const keyStyle = {
+        background: "gray",
+        borderRadius: "50%",
+        color: "white",
+      };
+      return (
+        <button style={keyStyle} onClick={() => this.props.onClick()}>
+          {this.props.children}
+        </button>
+      );
+    }
+  }
+
+  class Display extends React.Component {
+    render() {
+      const displayStyle = {
+        backgroundColor: "lightgray",
+        borderRadius: 5,
+        gridColumnEnd: "span 4",
+        fontSize: "2em",
+        fontFamily: "sans",
+        textAlign: "end",
+        padding: 5,
+      };
+
+      const { value, inputExponent } = this.props;
+      let display = value?.toString();
+
+      // The display should show a trailing decimal point
+      // and trailing zeroes if any have been entered.
+
+      // FIXME: This should instead be done with NumberFormat
+      // and its minimumFractionDigits option, but that is not
+      // yet implemented.
+      // FIXME: something weird about if-conditions: results
+      // are different when putting this expression directly
+      // in the if-statement
```

Comment:
The problem was I forgot some cases. I've added a hacky fix real quick that you can adjust as you see fit.